packages/components-react/projects/react-ssr-wrapper/tests/unit/specs/__snapshots__/react-ssr-wrapper.spec.tsx.snap

```diff
@@ -5720,12 +5721,297 @@ img {
+  font: normal normal 400 1rem/calc(6px + 2.125ex + 6px) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
+  margin-top: calc(-1 * (calc(6px + 2.125ex) + 10px + 2px * 2 + 8px * 2));
+  padding-top: calc(calc(6px + 2.125ex) + 10px + 2px * 2 + 8px * 2);
+  left: 0;
+  right: 0;
+  color: #6B6D70;
+}
+:host(:hover) {
+  color: #010205;
+}
+button {
+  position: absolute;
+  top: 0;
+  height: calc(calc(6px + 2.125ex) + 10px + 2px * 2 + 8px * 2);
+  width: 100%;
+  font: normal normal 400 1rem/calc(6px + 2.125ex + 6px) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
+  padding: 0;
+  background: transparent;
+  border: 2px solid #6B6D70;
+  border-radius: 4px;
+  outline: 0;
+  cursor: pointer;
+  transition: border-color var(--p-transition-duration, .24s) ease;
+}
+button:disabled {
+  cursor: not-allowed;
+  border-color: #949598;
+}
+ul {
+  display: flex;
+  flex-direction: column;
+  gap: 8px;
+  position: var(--p-internal-dropdown-position);
+  padding: 6px;
+  margin: 0;
+  background: #FFF;
+  font: normal normal 400 1rem/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
+  overflow-wrap: break-word;
+  hyphens: auto;
+  z-index: 10;
+  left: 0;
+  right: 0;
+  top: calc(100% - 2px);
+  opacity: 0;
+  overflow: hidden;
+  height: 1px;
+  pointer-events: none;
+  box-sizing: border-box;
+  max-height: 416px;
+  overflow-y: auto;
+  -webkit-overflow-scrolling: touch;
+  scroll-behavior: smooth;
+  border: 2px solid #6B6D70;
+  border-top: 1px solid #6B6D70 !important;
+  border-bottom-left-radius: 4px;
+  border-bottom-right-radius: 4px;
+  scrollbar-width: thin;
+  scrollbar-color: auto;
+  transition: border-color var(--p-transition-duration, .24s) ease;
+  transform: translate3d(0,0,0);
+}
+.option {
+  display: flex;
+  justify-content: space-between;
+  gap: 12px;
+  padding: 8px 12px;
+  flex: 1 0 calc(calc(6px + 2.125ex) + 8px * 2);
+  color: #535457;
+  cursor: pointer;
+  text-align: left;
+  word-break: break-word;
+  box-sizing: border-box;
+  border-radius: 4px;
+  transition: background-color var(--p-transition-duration, .24s) ease,color var(--p-transition-duration, .24s) ease;
+}
+.option[role=status] {
+  cursor: not-allowed;
+}
+.option__sr {
+  position: absolute;
+  width: 0;
+  height: 1px;
+  text-indent: -999999px;
+}
+.option:not([aria-disabled]):not([role=status]):hover {
+  color: #010205;
+  background: #EEEFF2;
+}
+.option--highlighted, .option--selected {
+  color: #010205;
+  background: #EEEFF2;
+}
+.option--selected {
+  cursor: default;
+  pointer-events: none;
+}
+.option--disabled {
+  cursor: not-allowed;
+  color: #949598;
+}
+.option--hidden {
+  display: none;
+}
+.optgroup {
+  display: block;
+  padding: 3px 14px;
+  font-weight: 600;
+}
+.optgroup:not(:first-child) {
+  margin-top: 8px;
+}
+.optgroup~.option {
+  padding-left: 24px;
+}
+.sr-text {
+  display: none;
+}
+          </style>
+          <button
```

Comment:
Let's add custom test cases for `filter` and `native` as well to cover the logic of the generated DSRComponent.